deploy/stacks/predeployment_stack.py

```diff
@@ -0,0 +1,67 @@
+from typing import Text
+from aws_cdk import (
+    core as cdk,   
+    aws_s3 as s3,
+    aws_cloudfront as cloudfront,
+    aws_ssm as ssm,
+    aws_route53 as route53,
+    aws_route53_targets as route53t,
+    aws_certificatemanager as acm
+)
+
+class PredeploymentStack(cdk.Stack):
+    """The stack is meant to deploy manually once before any other stack"""
+
+    def __init__(self, scope: cdk.Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Defining app constants
+        app_stage = self.node.try_get_context("app_stage")
+        props = self.node.try_get_context("props")
+
+        # --- Query deployment env specific config from SSM Parameter Store
+        hosted_zone_id = ssm.StringParameter.from_string_parameter_name(
+            self,
+            "HostedZoneID",
+            string_parameter_name="/hosted_zone/umccr/id"
+        ).string_value
+
+        hosted_zone_name = ssm.StringParameter.from_string_parameter_name(
+            self,
+            "HostedZoneName",
+            string_parameter_name="/hosted_zone/umccr/name"
+        ).string_value
+
+        # Fetch existing hosted_zone
+        hosted_zone = route53.HostedZone.from_hosted_zone_attributes(
```

Comment:
For some reason can't change this to just use Hosted Zone ID ([#Issue](https://github.com/aws/aws-cdk/issues/3663))